dags/prio/processor.py

```diff
@@ -0,0 +1,276 @@
+    task_id="load_processor_a",
+    source_bucket="moz-fx-data-prio-data",
+    source_object="staging/submission_date={{ ds }}/server_id=a/*",
+    destination_bucket="project-a-private",
+    destination_object="raw/submission_date={{ ds }}/",
+    google_cloud_storage_conn_id="google_cloud_prio_admin",
+    dag=dag,
+)
+
+load_processor_b = GoogleCloudStorageToGoogleCloudStorageOperator(
+    task_id="load_processor_b",
+    source_bucket="moz-fx-data-prio-data",
+    source_object="staging/submission_date={{ ds }}/server_id=b/*",
+    destination_bucket="project-b-private",
+    destination_object="raw/submission_date={{ ds }}/",
+    google_cloud_storage_conn_id="google_cloud_prio_admin",
+    dag=dag,
+)
+
+trigger_processor_a = GoogleCloudStorageToGoogleCloudStorageOperator(
+    task_id="trigger_processor_a",
+    source_bucket="moz-fx-data-prio-data",
+    source_object="staging/_SUCCESS",
+    destination_bucket="project-a-private",
+    destination_object="raw/_SUCCESS",
+    google_cloud_storage_conn_id="google_cloud_prio_admin",
+    dag=dag,
+)
+
+trigger_processor_b = GoogleCloudStorageToGoogleCloudStorageOperator(
+    task_id="trigger_processor_b",
+    source_bucket="moz-fx-data-prio-data",
+    source_object="staging/_SUCCESS",
+    destination_bucket="project-b-private",
+    destination_object="raw/_SUCCESS",
+    google_cloud_storage_conn_id="google_cloud_prio_admin",
+    dag=dag,
+)
+
+# Initiate the processors by spinning up GKE clusters and running the main
+# processing loop. These SubDags are coordinated by a higher level graph, but
+# they can be decoupled and instead run on the same schedule. This structure is
+# done for convenience, since both processors are running on the same
+# infrastructure (but partitioned along project boundaries).
+
+processor_a = SubDagOperator(
+    subdag=processing.prio_processor_subdag(
+        parent_dag_name=dag.dag_id,
+        child_dag_name="processor_a",
+        default_args=DEFAULT_ARGS,
+        server_id="a",
+        gcp_conn_id="google_cloud_prio_a",
+        service_account="prio-runner-a@moz-fx-priotest-project-a.iam.gserviceaccount.com",
+        arguments=["processor/bin/process"],
+        env_vars={
+            "DATA_CONFIG": "/app/processor/config/content.json",
+            "SERVER_ID": "A",
+            "SHARED_SECRET": "m/AqDal/ZSA9597GwMM+VA==",
+            # TODO: this is the built-in testing key
+            "PRIVATE_KEY_HEX": "{{ var.value.prio_private_key_hex_internal }}",
+            "PUBLIC_KEY_HEX_INTERNAL": "{{ var.value.prio_public_key_hex_internal }}",
+            "PUBLIC_KEY_HEX_EXTERNAL": "{{ var.value.prio_public_key_hex_external }}",
+            "BUCKET_INTERNAL_PRIVATE": "project-a-private",
+            "BUCKET_INTERNAL_SHARED": "project-a-shared",
+            "BUCKET_EXTERNAL_SHARED": "project-b-shared",
+            "RETRY_LIMIT": "5",
+            "RETRY_DELAY": "3",
+            "RETRY_BACKOFF_EXPONENT": "2",
```

Comment:
It looks like there is a substantial amount of duplicated code in this `env_vars` variable between processor a and b. Maybe you should define the common parts in a variable above and merge in whatever differs?

Reply:
If this were python 3, I would be less hesitant since I would be able to do this in place (`{**a, **b}`). The `RETRY_*` and `DATA_CONFIG` variables are the only shared variables, so it's not too bad here.